### PR TITLE
Remove git.io shortlinks from repo

### PR DESCRIPTION
**Pull request by Andoryuuta**
Github has deprecated git.io and will be [disabling all links on April 29th](https://github.blog/changelog/2022-04-25-git-io-deprecation/).
This PR replaces all git.io shortened links with the URL they redirect to.

| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Removed soon-to-be-invalid shortlinks
| Patch: Bug Fix?          | No
| Major: Breaking Change?  | No
| Minor: New Feature?      | No
| Tests Added + Pass?      | No
| Documentation PR Link    | N/A
| Any Dependency Changes?  | No
| License                  | MIT

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/14493"></a>

### CODE REVIEW

**Review by JLHwung — COMMENTED** [review]
Can you replace `master` to `main`?

File: packages/babel-core/src/parser/util/missing-plugin-helper.ts [hunk 1]
```diff
@@ -2,37 +2,37 @@ const pluginNameMap = {
...
       name: "@babel/plugin-syntax-async-do-expressions",
-      url: "https://github.com/babel/babel/tree/master/packages/babel-plugin-syntax-async-do-expressions",
+      url: "https://github.com/babel/babel/tree/main/packages/babel-plugin-syntax-async-do-expressions",
     },
   },
   classProperties: {
     syntax: {
       name: "@babel/plugin-syntax-class-properties",
-      url: "https://github.com/babel/babel/tree/master/packages/babel-plugin-syntax-class-properties",
+      url: "https://github.com/babel/babel/tree/main/packages/babel-plugin-syntax-class-properties",
```
Comment (SukkaW) [inline packages/babel-core/src/parser/util/missing-plugin-helper.ts:11]: The URL returns 404.

File: packages/babel-core/src/parser/util/missing-plugin-helper.ts [hunk 2]
```diff
@@ -2,37 +2,37 @@ const pluginNameMap = {
...
       name: "@babel/plugin-proposal-class-properties",
-      url: "https://github.com/babel/babel/tree/master/packages/babel-plugin-proposal-class-properties",
+      url: "https://github.com/babel/babel/tree/main/packages/babel-plugin-proposal-class-properties",
     },
   },
   classPrivateProperties: {
     syntax: {
       name: "@babel/plugin-syntax-class-properties",
-      url: "https://github.com/babel/babel/tree/master/packages/babel-plugin-syntax-class-properties",
+      url: "https://github.com/babel/babel/tree/main/packages/babel-plugin-syntax-class-properties",
```
Comment (SukkaW) [inline packages/babel-core/src/parser/util/missing-plugin-helper.ts:21]: The URL returns 404.

File: packages/babel-core/src/parser/util/missing-plugin-helper.ts [hunk 3]
```diff
@@ -2,37 +2,37 @@ const pluginNameMap = {
...
       name: "@babel/plugin-proposal-class-properties",
-      url: "https://github.com/babel/babel/tree/master/packages/babel-plugin-proposal-class-properties",
+      url: "https://github.com/babel/babel/tree/main/packages/babel-plugin-proposal-class-properties",
     },
   },
   classPrivateMethods: {
     syntax: {
       name: "@babel/plugin-syntax-class-properties",
-      url: "https://github.com/babel/babel/tree/master/packages/babel-plugin-syntax-class-properties",
+      url: "https://github.com/babel/babel/tree/main/packages/babel-plugin-syntax-class-properties",
```
Comment (SukkaW) [inline packages/babel-core/src/parser/util/missing-plugin-helper.ts:31]: The URL returns 404.

**Review by SukkaW — CHANGES_REQUESTED** [review]

**Review by JLHwung — APPROVED** [review]
The URLs returning 404 are archived plugins, we can fix them in another PR.

**Review by nicolo-ribaudo — APPROVED** [review]
Thanks! I agree, lets get this merged since it's quite urgent. We can iterate and improve the links later.

**Pull request merged**